### PR TITLE
Add delay to the redirect if its an error

### PR DESCRIPTION
**Pull request by prpcodes**
Added a delay of 5 seconds to the redirect if getting the request returns any error from the backend.

### CODE REVIEW

**Comment by Headscracher** [comment]
This seems to just work for direct errors thrown by trpc yet. Try calling an invalid or already completed request directly and you'll see in most cases you won't even get redirected yet, or redirected directly.

**Review by Headscracher — CHANGES_REQUESTED** [review]
This seems to just work for direct errors thrown by trpc yet. Try calling an invalid or already completed request directly and you'll see in most cases you won't even get redirected yet, or redirected directly.

**Comment by prpcodes** [comment]
> This seems to just work for direct errors thrown by trpc yet. Try calling an invalid or already completed request directly and you'll see in most cases you won't even get redirected yet, or redirected directly.

Thanks for the feedback. I refactored the code to ensure that the redirect only happens if the page is not found and the toast is closed automatically or by the user. It appears that there is a bug in react-toastify, in which the redirect does happen immediately if [_reactStrictMode_ is set to _true_](https://github.com/fkhadra/react-toastify/issues/741#issuecomment-1142595079), but this obviously should not occur in production, so we should be fine.

File: src/pages/playground/request/[id].tsx [hunk 1]
```diff
@@ -43,9 +44,14 @@ const PlaygroundRequest: PageWithLayout = ({}) => {
     router.query.id as string,
     {
       retry: 1,
-      onError: () => {
-        toast.error('The request could not be found');
-        void router.push('/playground', undefined, { shallow: true });
+      onError: (e) => {
+        if (e.message === 'NOT_FOUND') {
```
Comment (MatsThieme) [inline src/pages/playground/request/[id].tsx:48]: what about any other error message? should there be an else block?

Comment (MatsThieme) [reply src/pages/playground/request/[id].tsx:48]: didnt read the previous conversation.

**Review by MatsThieme — APPROVED** [review]
👍

**Review by Headscracher — APPROVED** [review]

**Pull request merged**